wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -1669,6 +1798,137 @@ private function execute_use_statement( WP_Parser_Node $node ): void {
+		 *
+		 * The grammar is non-trivial, and supports multi-statements like:
+		 *   SET @var = '...', SESSION sql_mode = '...', @@GLOBAL.time_zone = '...', @@debug = '...', ...
+		 *
+		 * This will be flattened into a single array of grammar node lists:
+		 *   [
+		 *     [ <userVariable>, <equal>, <expr> ],
+		 *     [ <optionType>, <internalVariableName>, <equal>, <setExprOrDefault> ],
+		 *     [ <setSystemVariable>, <equal>, <setExprOrDefault> ],
+		 *     [ <setSystemVariable>, <equal>, <setExprOrDefault> ],
+		 *   ]
+		 */
+		$subnode = $node->get_first_child_node();
+		if ( $subnode->has_child_node( 'optionValueNoOptionType' ) ) {
+			$start_node  = $subnode->get_first_child_node( 'optionValueNoOptionType' );
+			$definitions = array( $start_node->get_children() );
+		} elseif ( $subnode->has_child_node( 'startOptionValueListFollowingOptionType' ) ) {
+			$start_node  = $subnode
+				->get_first_child_node( 'startOptionValueListFollowingOptionType' )
+				->get_first_child_node( 'optionValueFollowingOptionType' ) ?? $node;
+			$definitions = array(
+				array_merge(
+					array( $subnode->get_first_child_node( 'optionType' ) ),
+					$start_node->get_children()
+				),
+			);
+		} else {
+			$definitions = array( $subnode->get_children() );
+		}
+
+		$continue_node = $subnode->get_first_child_node( 'optionValueListContinued' );
+		if ( $continue_node ) {
+			foreach ( $continue_node->get_child_nodes( 'optionValue' ) as $child ) {
+				$node          = $child->get_first_child_node( 'optionValueNoOptionType' ) ?? $child;
+				$definitions[] = $node->get_child_nodes();
+			}
+		}
+
+		/*
+		 * 2. Iterate and process the SET definitions.
+		 *
+		 * When an "optionType" node is encountered (such as "SESSION var = ..."),
+		 * it's value is used for all following system variable assignments that
+		 * have no type keyword specified, until the next "optionType" is found.
+		 *
+		 * This doesn't apply to "@@" type prefixes (such as "@@SESSION.var_name"),
+		 * which always impact only the immediately following system variable.
+		 */
+		$default_type = WP_MySQL_Lexer::SESSION_SYMBOL;
+		foreach ( $definitions as $definition ) {
+			// Check if the definition starts with an "optionType" node with
+			// one of the SESSION, GLOBAL, PERSIST, or PERSIST_ONLY tokens.
+			$part = array_shift( $definition );
+			if ( $part instanceof WP_Parser_Node && 'optionType' === $part->rule_name ) {
+				$default_type = $part->get_first_child_token()->id;
+				$part         = array_shift( $definition );
+			}
+
+			if (
+				$part instanceof WP_Parser_Node
+				&& (
+					'internalVariableName' === $part->rule_name
+					|| 'setSystemVariable' === $part->rule_name
+				)
+			) {
+				array_shift( $definition ); // Remove the '='.
+				$value = array_shift( $definition );
+				$this->execute_set_system_variable_statement( $part, $value, $default_type );
+			}
+		}
+
+		// TODO: Support user variables (in-memory or a temporary table).
```

Comment:
let's communicate lack of support to at least make the developer aware why their code breaks

Reply:
👍 Added in https://github.com/Automattic/sqlite-database-integration/pull/29/commits/d6f6b6d93c94d0e181a89677b75c8a2284c790aa.